PGM/src/main/java/tc/oc/pgm/mutation/types/kit/TeamChestMutation.java

```
@@ -0,0 +1,135 @@
+        if (event.getCurrentItem() == null) return;
+
+        // No putting evil items (ender chest, possibly wool) into the chest
+        Optional<Inventory> teamChest = getTeamsInventory(event.getActor());
+        if (teamChest.filter(teamInventory -> {
+            return teamInventory.equals(event.getView().getTopInventory());
+        }).isPresent() &&
+                isItemEvil(event.getCurrentItem())) {
+            event.setCancelled(true);
+            return;
+        }
+
+        // If normal right click, in their inventory, on the chest, then open shared inventory.
+        getTeamsInventory(event.getActor()).ifPresent(teamInventory -> {
+            if (event.getInventory().getType() == InventoryType.CRAFTING &&
+                    event.getCurrentItem().getType() == TOOL_TYPE &&
+                    event.getAction() == InventoryAction.PICKUP_HALF) {
+                event.setCancelled(true);
+                // This resets their mouse position annoyingly, but without it items can get stuck in places.
+                event.getActor().closeInventory();
+                // Prevent visual inconsistencies, specifically off-hand slot
+                if (event.getSlotType() == InventoryType.SlotType.QUICKBAR) {
+                    event.getActor().updateInventory();
+                }
+                event.getActor().openInventory(teamInventory);
+            }
+        });
+    }
+
+    private boolean isItemEvil(ItemStack item) {
+        if(item.getType() == TOOL_TYPE) return true;
+        if(oWmm.filter(wmm -> wmm.isObjectiveWool(item)).isPresent()) return true;
+
+        return false;
+    }
+
+    private Optional<Inventory> getTeamsInventory(Player bukkitPlayer) {
+        MatchPlayer player = match().getPlayer(bukkitPlayer);
+        Party team = player.getParty();
+
+        if (!team.isParticipating()) return Optional.empty();
+
+        return Optional.of(teamChests.get(team));
```

Can `teamChests.get(team)` be null? If so you want `Optional.ofNullable` or else a runtime error will be thrown.

---

"""
I don't think you need to throw an exception if a chest isn't found, can't think of any cases where that should happen.
"""
- Shiny